### PR TITLE
adding metadata support, refactoring, implemement destroy

### PR DESCRIPTION
**Pull request by dgraeber**
*Issue #, if available:*

*Description of changes:*

- Added support for storing metadata and using that metadata as input
- Implemented destroy
- Refactored code structure
- Fixed typos in pydocs

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: seedfarmer/cli_groups/_test_group.py [hunk 1]
```diff
@@ -90,25 +92,41 @@ def test() -> None:
     multiple=True,
     required=False,
 )
+@click.option(
+    "--metadata-path",
+    "-mp",
+    default=None,
+    help="""The relative path (to seedfarmer.yaml) of the JSON document that represents
+     the module metadata to ber used in deploying.
```
Comment (kukushking) [inline seedfarmer/cli_groups/_test_group.py:100]: typo

File: seedfarmer/mgmt/module_info.py [hunk 2]
```diff
@@ -737,10 +792,33 @@ def remove_deployed_deployment_manifest(deployment: str, session: Optional[Sessi
     ssm.delete_parameters(parameters=[_deployed_deployment_manifest_key(deployment)], session=session)
 
 
+def remove_testmetatainput(deployment: str, group: str, module: str, session: Optional[Session] = None) -> None:
```
Comment (malachi-constant) [inline seedfarmer/mgmt/module_info.py:795]: can we rename this to `remove_test_metadata_input()` and then follow snakecase for the other metadata functions?

File: seedfarmer/cli_groups/_test_group.py (hunk 1 above)
Comment (dgraeber) [reply seedfarmer/cli_groups/_test_group.py:100]: on it

File: seedfarmer/mgmt/module_info.py (hunk 2 above)
Comment (dgraeber) [reply seedfarmer/mgmt/module_info.py:795]: yes

**Pull request merged**